hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/helpers/SnapshotInfo.java

```diff
@@ -586,15 +688,24 @@ public boolean equals(Object o) {
         Objects.equals(
             globalPreviousSnapshotId, that.globalPreviousSnapshotId) &&
         snapshotPath.equals(that.snapshotPath) &&
-        checkpointDir.equals(that.checkpointDir);
+        checkpointDir.equals(that.checkpointDir) &&
+        deepClean == that.deepClean &&
```

Comment:
I was wondering if we need to add `deepClean` and `sstFiltered` for `equals` as these can change depending on the moment if it was deep cleaned or sst filtered. I guess it should be fine.

Reply:
Depending on how `SnapshotInfo` is used in comparisons.

We could deliberately ignore them if it fits the use case.

If not I would keep it here for completeness.